src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -252,4 +300,12 @@ private Path createSchemedPath(Path p) {
         }
         return p;
     }
+
+    private FileStatus convertFileStatusToSrcPath(FileStatus fileStatus, PathProperties pathProperties) {
+        String mappedUri = fileStatus.getPath().toString();
+        String srcUri = mappedUri.replaceFirst(pathProperties.getDstPrefix(), pathProperties.getSrcPrefix());
```

Comment:
This only works when everything is going as planned.  Because this depends also on everything else in the Hadoop ecosystem (and not just on ourselves), I would rather only replace the dst prefix when it is indeed a prefix.  For instance first verify we're doing well
```java
if (!mappedUri.startsWith(pathProperties.getDstPrefix())) {
  throw new SomeLogicIsBrokenHereOrInHadoopException(...);
  // or maybe just log nd return fileStatus, to keep going as though nothing went wrong.
}
```

Reply:
true